hyperlink/test/test_decoded_url.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from .. import DecodedURL
+from .._url import _percent_decode
+from .common import HyperlinkTestCase
+
+BASIC_URL = 'http://example.com/#'
+TOTAL_URL = "https://%75%73%65%72:%00%00%00%00@xn--bcher-kva.ch:8080/a/nice%20nice/./path/?zot=23%25&zut#frég"
+
+
+class TestURL(HyperlinkTestCase):
+
+    def test_durl_basic(self):
+        bdurl = DecodedURL.from_text(BASIC_URL)
+        assert bdurl.scheme == 'http'
+        assert bdurl.host == 'example.com'
+        assert bdurl.port == 80
+        assert bdurl.path == ('',)
+        assert bdurl.fragment == ''
+
+        durl = DecodedURL.from_text(TOTAL_URL)
+
+        assert durl.scheme == 'https'
+        assert durl.host == 'bücher.ch'
+        assert durl.port == 8080
+        assert durl.path == ('a', 'nice nice', '.', 'path', '')
+        assert durl.fragment == 'frég'
+        assert durl.get('zot') == ['23%']
+
+        assert durl.user == 'user'
+        assert durl.userinfo == ('user', '\0\0\0\0')
+
+    def test_passthroughs(self):
```

Comment:
A lot of these (e.g., `sibling`, `__eq__`) aren't passthroughs.  A better name for it would be `test_interface_equivalent_to_URL`.